pom.xml

```diff
@@ -891,6 +905,41 @@ under the License.
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-dependency-plugin</artifactId>
+				<version>3.0.2</version>
+				<executions>
+					<execution>
+						<id>analyze</id>
+						<goals>
+							<goal>analyze-only</goal>
+						</goals>
+						<configuration>
+							<!-- Ignore runtime-only dependencies in analysis -->
+							<ignoreNonCompile>true</ignoreNonCompile>
+							<outputXML>true</outputXML>
+							<failOnWarning>true</failOnWarning>
+							<ignoredUnusedDeclaredDependencies>
```

Comment:
Would it be better to move the dependencies out of the parent pom?

Reply:
I would keep logging, junit, jsr305 and force-shading in the parent pom as they are used by virtually all modules and would just clutter up other poms.

But I wouldn't mind moving the mocking/hamcrest dependencies to child poms.

Reply:
Can we keep the root dependencies? I find the current way actually convenient and avoiding pom clutter.

As Chesnay mentioned, the compile scope dependencies should be okay (used everywhere).

Could the test dependencies be separately skipped?